### PR TITLE
Add regex logging filter

### PR DESCRIPTION
**Pull request by StephanErb**
Main contribution of this pull request is the addition of a `RegexFilter` that can be used to prevent log messages from being emitted if those match a regular expression. This is useful for common requests like `/health` and `/metrics` that otherwise spam the logs significantly.

Additional changes (and yak shaving):

* Add migration helper to use the new `RegexFilter` also within the legacy Python `fileConfig` logger configuration format.
* Drop pyscaffold and Python 2.6 support
* Perform minor code cleanups and sanitisation of the generated output

For the best review experience, I would suggest to look at each commit individually rather than the entire PR diff.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,526 +1,29 @@
...
+    license='new BSD',
+    classifiers=[
+        'Development Status :: 4 - Beta',
+        'Environment :: Console',
+        'Intended Audience :: Developers',
+        'Intended Audience :: System Administrators',
+        'License :: OSI Approved :: BSD License',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3.4',
```
Comment (theMarix) [inline setup.py:27]: You should probably also add the classifier for Python 3.6 as you added that one to the test.

File: cee_syslog_handler/__init__.py [hunk 2]
```diff
@@ -48,16 +48,9 @@ def get_full_message(exc_info, message):
...
-def make_message_dict(record, debugging_fields, extra_fields, fqdn, localname, facility):
-    if fqdn:
-        host = socket.getfqdn()
-    elif localname:
-        host = localname
-    else:
-        host = socket.gethostname()
+def make_message_dict(record, debugging_fields, extra_fields, facility):
     message_dict = {
-        'version': "1.0",
```
Comment (theMarix) [inline cee_syslog_handler/__init__.py:64]: Do we know _why_ this was there?

File: cee_syslog_handler/__init__.py [hunk 3]
```diff
@@ -232,3 +232,45 @@ def format(self, log_record):
...
+        Returns True if the record shall be logged. False otherwise.
+
+        https://github.com/python/cpython/blob/2.7/Lib/logging/__init__.py#L607
+        """
+        return self._regex_filter.filter(record) and super(FilteredNamedCeeLogger, self).filter(record)
+
+
+class RegexFilter(logging.Filter):
+    """
+    This Filter can be used to discard log messages that match a given regular expression.
```
Comment (theMarix) [inline cee_syslog_handler/__init__.py]: The way this is documented it sounds like regexes would be matched to the string using `re.match`, that is, they would only match at the beginning of the message. However, the way this is written `re.search` is used, which will match anywhere in the message. We should probably either change the documentation – e.g. to _contains a match_ – or adjust the implementation if the start-of-message behaviour is the actually desired one.

**Review by theMarix — CHANGES_REQUESTED** [review]
Thanks for the PR. Overall I like it very, just some details to clean up or clarify. :)

File: cee_syslog_handler/__init__.py (hunk 2 above)
Comment (StephanErb) [reply cee_syslog_handler/__init__.py:64]: [The cee_syslog handler is 90% copy past from `graypy` that logs in GELF format](https://github.com/hoffmann/graypy/blob/master/graypy/handler.py#L90). GELF requires a version string but CEE does not. I believe this was now just unnecessary copy paste.

Comment (theMarix) [reply cee_syslog_handler/__init__.py:64]: 👍

File: setup.py (hunk 1 above)
Comment (StephanErb) [reply setup.py:27]: Added. Thanks.

File: cee_syslog_handler/__init__.py (hunk 3 above)
Comment (StephanErb) [reply cee_syslog_handler/__init__.py]: Good catch. I have updated the docs.

**Review by theMarix — APPROVED** [review]

**Pull request merged**